sdk/python/kfp/compiler/_container_builder.py

```diff
@@ -18,19 +18,37 @@
 import os
 import uuid
 
+SERVICEACCOUNT_NAMESPACE = '/var/run/secrets/kubernetes.io/serviceaccount/namespace'
+GCS_STAGING_BUCKET_DEFAULT = 'gs://kfp_container_build_staging'
+
 class ContainerBuilder(object):
   """
   ContainerBuilder helps build a container image
   """
-  def __init__(self, gcs_staging, namespace):
+  def __init__(self, gcs_staging=GCS_STAGING_BUCKET_DEFAULT, namespace=None):
     """
     Args:
-      gcs_staging (str): GCS blob that can store temporary build files
+      gcs_staging (str): GCS bucket/blob that can store temporary build files,
+          default is GCS_STAGING_BUCKET_DEFAULT defined above.
+      namespace (str): kubernetes namespace where the pod is launched,
+          default is the same namespace as the notebook service account in cluster
+              or 'kubeflow' if not in cluster
     """
     if not gcs_staging.startswith('gs://'):
       raise ValueError('Error: {} should be a GCS path.'.format(gcs_staging))
     self._gcs_staging = gcs_staging
+    from ._gcs_helper import GCSHelper
+    from pathlib import PurePath
+    gcs_bucket = PurePath(self._gcs_staging).parts[1]
```

Comment:
check parts length before reference it?

Reply:
good point

Reply:
Although I've checked that it starts with gs://, which makes the length more than 1.
I'll add more checks here.